In [1]:
import GEOparse

print("GEOparse imported successfully")
print("Version:", GEOparse.__version__)


GEOparse imported successfully
Version: 2.0.4


In [2]:
gse_axolotl = GEOparse.get_GEO(
    geo="GSE106269",
    destdir="../data/axolotl/metadata"
)

print("GSE106269 loaded successfully")

04-Sep-2026 22:11:06 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/soft/GSE106269_family.soft.gz to ../data/axolotl/metadata/GSE106269_family.soft.gz
100%|█████████████████████████████████████████████████████████████████████████████| 83.1k/83.1k [00:02<00:00, 42.4kB/s]
04-Sep-2026 22:11:10 DEBUG downloader - Size validation passed
04-Sep-2026 22:11:10 DEBUG downloader - Moving /tmp/tmpvq9ok_7d to /home/ghayyas/FYP_Regeneration/data/axolotl/metadata/GSE106269_family.soft.gz
04-Sep-2026 22:11:10 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/soft/GSE106269_family.soft.gz
04-Sep-2026 22:11:10 INFO GEOparse - Parsing ../data/axolotl/metadata/GSE106269_family.soft.gz: 
04-Sep-2026 22:11:10 DEBUG GEOparse - DATABASE: GeoMiame
04-Sep-2026 22:11:10 DEBUG GEOparse - SERIES: GSE106269
04-Sep-2026 22:11:10 DEBUG GEOparse - PLATFORM: GPL22800
04-Sep-2026 22:11:10 DEBUG GEOparse - SAMPLE: GSM3204856
04-Sep

GSE106269 loaded successfully


In [3]:
print(gse_axolotl.metadata['title'])
print(gse_axolotl.metadata['summary'])

['Single-cell analysis uncovers convergence of cell identities during axolotl limb regeneration']
['Amputation of the axolotl forelimb results in the formation of a blastema, a transient tissue where progenitor cells accumulate prior to limb regeneration. Connective tissue (CT) – skeleton, periskeleton, tendon, dermis, interstitial cells – contributes the vast majority of cells that populate the blastema, however it is unclear how individual CT cells may reprogram their fate in order to rebuild the tetrapod limb. Here we use a combination of Cre-loxP reporter lineage tracking and single-cell (sc) RNA-seq to molecularly track, for the first time, adult CT cell heterogeneity and its transition to a limb blastema state. We uncover a multi-phasic molecular program where CT cell types found in the uninjured adult limb revert to a relatively homogenous progenitor state that participates in inflammation and extracellular matrix disassembly prior to proliferation, establishment of positional i

In [4]:
print(f"Number of samples: {len(gse_axolotl.gsms)}")

for gsm_name, gsm in gse_axolotl.gsms.items():
    print(gsm_name, "-", gsm.metadata['title'][0])

Number of samples: 1604
GSM3204856 - axo2.A1
GSM3204857 - axo2.A10
GSM3204858 - axo2.A12
GSM3204859 - axo2.A2
GSM3204860 - axo2.A3
GSM3204861 - axo2.A4
GSM3204862 - axo2.A7
GSM3204863 - axo2.B1
GSM3204864 - axo2.B11
GSM3204865 - axo2.B3
GSM3204866 - axo2.B7
GSM3204867 - axo2.C1
GSM3204868 - axo2.C2
GSM3204869 - axo2.C3
GSM3204870 - axo2.C4
GSM3204871 - axo2.D10
GSM3204872 - axo2.D11
GSM3204873 - axo2.D12
GSM3204874 - axo2.D4
GSM3204875 - axo2.D6
GSM3204876 - axo2.D8
GSM3204877 - axo2.D9
GSM3204878 - axo2.E1
GSM3204879 - axo2.E11
GSM3204880 - axo2.E4
GSM3204881 - axo2.E6
GSM3204882 - axo2.E7
GSM3204883 - axo2.F1
GSM3204884 - axo2.G1
GSM3204885 - axo2.G10
GSM3204886 - axo2.G3
GSM3204887 - axo2.G4
GSM3204888 - axo2.G6
GSM3204889 - axo2.G8
GSM3204890 - axo2.H1
GSM3204891 - axo2.H12
GSM3204892 - axo2.H5
GSM3204893 - axo2.H6
GSM3204894 - axo3.A1
GSM3204895 - axo3.A11
GSM3204896 - axo3.A12
GSM3204897 - axo3.A3
GSM3204898 - axo3.A4
GSM3204899 - axo3.A5
GSM3204900 - axo3.A6
GSM3204901 - axo3.A7

In [5]:
for gsm_name, gsm in list(gse_axolotl.gsms.items())[:20]:
    print(gsm_name, "-", gsm.metadata['title'][0])
    

GSM3204856 - axo2.A1
GSM3204857 - axo2.A10
GSM3204858 - axo2.A12
GSM3204859 - axo2.A2
GSM3204860 - axo2.A3
GSM3204861 - axo2.A4
GSM3204862 - axo2.A7
GSM3204863 - axo2.B1
GSM3204864 - axo2.B11
GSM3204865 - axo2.B3
GSM3204866 - axo2.B7
GSM3204867 - axo2.C1
GSM3204868 - axo2.C2
GSM3204869 - axo2.C3
GSM3204870 - axo2.C4
GSM3204871 - axo2.D10
GSM3204872 - axo2.D11
GSM3204873 - axo2.D12
GSM3204874 - axo2.D4
GSM3204875 - axo2.D6


In [6]:
gsm = next(iter(gse_axolotl.gsms.values()))

print(gsm.metadata.keys())

dict_keys(['title', 'geo_accession', 'status', 'submission_date', 'last_update_date', 'type', 'channel_count', 'source_name_ch1', 'organism_ch1', 'taxid_ch1', 'characteristics_ch1', 'treatment_protocol_ch1', 'molecule_ch1', 'extract_protocol_ch1', 'description', 'data_processing', 'platform_id', 'contact_name', 'contact_email', 'contact_laboratory', 'contact_department', 'contact_institute', 'contact_address', 'contact_city', 'contact_zip/postal_code', 'contact_country', 'instrument_model', 'library_selection', 'library_source', 'library_strategy', 'relation', 'supplementary_file_1', 'series_id', 'data_row_count'])


In [7]:
for key, value in gsm.metadata.items():
    if key in ['title', 'source_name_ch1', 'characteristics_ch1',
               'platform_id', 'instrument_model', 'library_strategy',
               'supplementary_file_1']:
        print(f"\n{key}:")
        print(value)


title:
['axo2.A1']

source_name_ch1:
['Stage 44 forelimb limb bud']

characteristics_ch1:
['sc-rna-seq method: Fluidigm C1', 'development stage: stage 44 limb bud']

platform_id:
['GPL22800']

instrument_model:
['Illumina HiSeq 2500']

library_strategy:
['RNA-Seq']

supplementary_file_1:
['NONE']


In [8]:
from collections import Counter

stage_counts = Counter()

for gsm in gse_axolotl.gsms.values():
    for characteristic in gsm.metadata.get('characteristics_ch1', []):
        stage_counts[characteristic] += 1

for characteristic, count in stage_counts.most_common():
    print(count, "-", characteristic)

1127 - sc-rna-seq method: Fluidigm C1
940 - cre line: Prrx1:Cre-ER
467 - sc-rna-seq method: SmartSeq2
412 - development stage: blastema 11 dpa
385 - cre line: Col1a2:ER-Cre-ER
167 - development stage: blastema 5 dpa
144 - development stage: uninjured upper arm
135 - development stage: blastema 18 dpa
121 - development stage: stage 44 limb bud
121 - development stage: blastema 8 dpa
108 - development stage: blastema 3 dpa
82 - development stage: stage 28 limb field
77 - development stage: uninjured upper arm 3mpa
76 - development stage: stage 40 limb bud
51 - development stage: uninjured upper arm 1apa
10 - sc-rna-seq method: 10X Genomics
10 - tissue: uninjured upper arm section


In [9]:
from collections import Counter

platform_counts = Counter()

for gsm in gse_axolotl.gsms.values():
    platform = gsm.metadata.get('platform_id', ['Unknown'])[0]
    platform_counts[platform] += 1

for platform, count in platform_counts.most_common():
    print(count, "-", platform)

1604 - GPL22800


In [10]:
print("Series supplementary files:")
print(gse_axolotl.metadata.get('supplementary_file', 'none listed'))

Series supplementary files:
['ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/suppl/GSE106269_RAW.tar', 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/suppl/GSE106269_Table_S3.csv.gz', 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/suppl/GSE106269_Table_S5.csv.gz', 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/suppl/GSE106269_Table_S7.csv.gz', 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/suppl/GSE106269_Table_S8.csv.gz', 'ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE106nnn/GSE106269/suppl/GSE106269_Table_S9.csv.gz']


In [11]:
print("\nExample per-sample supplementary files:")

shown = 0
for gsm_name, gsm in gse_axolotl.gsms.items():
    files = gsm.metadata.get('supplementary_file_1', [])
    
    if files and files != ['NONE']:
        print(gsm_name, "-", files)
        shown += 1
    
    if shown >= 10:
        break


Example per-sample supplementary files:
GSM3206450 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206450/suppl/GSM3206450_GER006_10x_barcodes.tsv.gz']
GSM3206451 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206451/suppl/GSM3206451_GER007_10x_barcodes.tsv.gz']
GSM3206452 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206452/suppl/GSM3206452_GER014_10x_barcodes.tsv.gz']
GSM3206453 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206453/suppl/GSM3206453_GER015_10x_barcodes.tsv.gz']
GSM3206454 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206454/suppl/GSM3206454_GER017_10x_barcodes.tsv.gz']
GSM3206455 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206455/suppl/GSM3206455_GER018_10x_barcodes.tsv.gz']
GSM3206456 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206456/suppl/GSM3206456_GER019_10x_barcodes.tsv.gz']
GSM3206457 - ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3206nnn/GSM3206457/suppl/GSM3206457_GER020_10x_b

In [12]:
gse_mouse = GEOparse.get_GEO(
    geo="GSE131078",
    destdir="../data/mouse/metadata"
)

print("GSE131078 loaded successfully")
print("Number of samples:", len(gse_mouse.gsms))

04-Sep-2026 22:18:51 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE131nnn/GSE131078/soft/GSE131078_family.soft.gz to ../data/mouse/metadata/GSE131078_family.soft.gz
100%|█████████████████████████████████████████████████████████████████████████████| 3.45k/3.45k [00:01<00:00, 2.49kB/s]
04-Sep-2026 22:18:54 DEBUG downloader - Size validation passed
04-Sep-2026 22:18:54 DEBUG downloader - Moving /tmp/tmphczc9scj to /home/ghayyas/FYP_Regeneration/data/mouse/metadata/GSE131078_family.soft.gz
04-Sep-2026 22:18:54 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE131nnn/GSE131078/soft/GSE131078_family.soft.gz
04-Sep-2026 22:18:54 INFO GEOparse - Parsing ../data/mouse/metadata/GSE131078_family.soft.gz: 
04-Sep-2026 22:18:54 DEBUG GEOparse - DATABASE: GeoMiame
04-Sep-2026 22:18:54 DEBUG GEOparse - SERIES: GSE131078
04-Sep-2026 22:18:54 DEBUG GEOparse - PLATFORM: GPL21493
04-Sep-2026 22:18:54 DEBUG GEOparse - SAMPLE: GSM3763054
04-Sep-2026 

GSE131078 loaded successfully
Number of samples: 24


In [13]:
sample_name = list(gse_mouse.gsms.keys())[0]
gsm = gse_mouse.gsms[sample_name]

print("Sample:", sample_name)
print("Title:", gsm.metadata.get("title"))
print("Source:", gsm.metadata.get("source_name_ch1"))
print("Characteristics:", gsm.metadata.get("characteristics_ch1"))
print("Organism:", gsm.metadata.get("organism_ch1"))
print("Platform:", gsm.metadata.get("platform_id"))
print("Instrument:", gsm.metadata.get("instrument_model"))
print("Library strategy:", gsm.metadata.get("library_strategy"))
print("Supplementary file:", gsm.metadata.get("supplementary_file_1"))

Sample: GSM3763054
Title: ['R12_1']
Source: ['12 DPA, distal amputation']
Characteristics: ['strain: C57BL/6', 'age: 10-12 weeks old', 'tissue: Hind limb digit tip (bone, cartilage, marrow, dermis, epidermis)', 'treatment: distal amputation', 'time: 12 DPA']
Organism: ['Mus musculus']
Platform: ['GPL21493']
Instrument: ['Illumina HiSeq 3000']
Library strategy: ['RNA-Seq']
Supplementary file: ['NONE']


In [14]:
for gsm_name, gsm in gse_mouse.gsms.items():
    print(
        gsm_name,
        "|",
        gsm.metadata.get("title", [""])[0],
        "|",
        gsm.metadata.get("source_name_ch1", [""])[0],
        "|",
        "; ".join(gsm.metadata.get("characteristics_ch1", []))
    )

GSM3763054 | R12_1 | 12 DPA, distal amputation | strain: C57BL/6; age: 10-12 weeks old; tissue: Hind limb digit tip (bone, cartilage, marrow, dermis, epidermis); treatment: distal amputation; time: 12 DPA
GSM3763055 | R12_2 | 12 DPA, distal amputation | strain: C57BL/6; age: 10-12 weeks old; tissue: Hind limb digit tip (bone, cartilage, marrow, dermis, epidermis); treatment: distal amputation; time: 12 DPA
GSM3763056 | R12_3 | 12 DPA, distal amputation | strain: C57BL/6; age: 10-12 weeks old; tissue: Hind limb digit tip (bone, cartilage, marrow, dermis, epidermis); treatment: distal amputation; time: 12 DPA
GSM3763057 | R12_4 | 12 DPA, distal amputation | strain: C57BL/6; age: 10-12 weeks old; tissue: Hind limb digit tip (bone, cartilage, marrow, dermis, epidermis); treatment: distal amputation; time: 12 DPA
GSM3763058 | R14_1 | 14 DPA, distal amputation | strain: C57BL/6; age: 10-12 weeks old; tissue: Hind limb digit tip (bone, cartilage, marrow, dermis, epidermis); treatment: distal 

In [15]:
print("Series supplementary files:")
print(gse_mouse.metadata.get('supplementary_file', 'none listed'))

print("\nExample per-sample supplementary files:")

shown = 0
for gsm_name, gsm in gse_mouse.gsms.items():
    files = gsm.metadata.get('supplementary_file_1', [])
    
    if files and files != ['NONE']:
        print(gsm_name, "-", files)
        shown += 1
    
    if shown >= 10:
        break

Series supplementary files:
['ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE131nnn/GSE131078/suppl/GSE131078_gene_counts.txt.gz']

Example per-sample supplementary files:


### GSE106269 — Axolotl

GSE106269 contains 1,604 GEO sample records from the axolotl limb regeneration study by Gerber et al. The dataset contains multiple single-cell RNA-seq experimental subsets and platforms, including Fluidigm C1, SmartSeq2, and 10X Genomics. Sample metadata identify developmental limb-bud stages, uninjured tissue, and multiple post-amputation blastema timepoints. Series-level supplementary files include the raw data archive (`GSE106269_RAW.tar`) and several supplementary study tables, while individual 10X samples have sample-level barcode files. The dataset will therefore require careful subset selection and metadata-based identification before downstream scRNA-seq analysis.

### GSE131078 — Mouse

GSE131078 contains 24 bulk RNA-seq samples from mouse hind-limb digit tips, consisting of 12 regenerative (distal amputation) and 12 non-regenerative (proximal amputation) samples. Each condition contains four biological replicates at 12, 14, and 21 days post-amputation (DPA), giving a balanced 2 × 3 × 4 experimental structure. GEO metadata clearly identify treatment, timepoint, tissue, strain, and sequencing information. A single series-level processed gene-count matrix (`GSE131078_gene_counts.txt.gz`) is available and no individual sample supplementary files are listed. This processed matrix will be used for downstream bulk RNA-seq analysis after its structure and gene/sample identifiers are verified.
